forte/data/ontology/core.py

```diff
@@ -446,30 +408,33 @@ class FNdArray:
     """
 
     def __init__(
-        self, dtype: Optional[str] = None, shape: Optional[List[int]] = None
+        self,
+        dtype: Optional[str] = None,
+        shape: Optional[List[int]] = None,
+        data_ref: Optional[List] = None,
     ):
         super().__init__()
-        self._dtype: Optional[np.dtype] = (
-            np.dtype(dtype) if dtype is not None else dtype
+        self.__data_ref: List = (
+            [dtype, shape, None] if data_ref is None else data_ref
         )
-        self._shape: Optional[tuple] = (
-            tuple(shape) if shape is not None else shape
-        )
-        self._data: Optional[np.ndarray] = None
-        if dtype and shape:
-            self._data = np.ndarray(shape, dtype=dtype)
 
     @property
     def dtype(self):
-        return self._dtype
+        dtype = self.__data_ref[0]
+        return np.dtype(dtype) if dtype is not None else dtype
```

Comment:
`else dtype` simply returns `None`?

Reply:
Yes. This is consistent with the previous behavior 
```python
self._dtype: Optional[np.dtype] = (
    np.dtype(dtype) if dtype is not None else dtype
)
```